### PR TITLE
feat: improve single request proxy

### LINKED ISSUE #106: Deploy new Single Request Proxy Subgraphs

### PR DESCRIPTION
**Pull request by aimensahnoun**
Resolve #106 
### Example of data in the subgraph

image

## Release Notes

- **New Features**
	- Enhanced event emissions for `SingleRequestProxyFactory` with additional fee-related parameters (`feeAddress`, `feeAmount`, `feeProxyUsed`) in the `ERC20SingleRequestProxyCreated` and `EthereumSingleRequestProxyCreated` events.
	- Introduced new data source configurations for `SingleRequestProxyFactory` across multiple networks, including Ethereum, Avalanche, Binance Smart Chain, and others.
	- New entity `SingleRequestProxyDeployment` added to track deployments related to the `SingleRequestProxyFactory`.

- **Bug Fixes**
	- Updated contract address and start block for `SingleRequestProxyFactory` in the Sepolia network configuration.

These updates improve the tracking and management of single request proxies within the application.

### CODE REVIEW

**Comment by aimensahnoun** [comment]
### Example of data in the subgraph

image

File: schema.graphql [hunk 1]
```diff
@@ -81,6 +81,9 @@ type SingleRequestProxyDeployment @entity {
   tokenAddress: Bytes
   paymentReference: Bytes!
   proxyType: SingleRequestProxyType!
+  feeAddress: Bytes!
+  feeAmount: BigInt!
+  feeProxyUsed: Bytes!
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply schema.graphql:86]: Unnecessary. Resolving as No Problem :ok_person:

File: subgraph.avalanche.yaml [hunk 2]
```diff
@@ -153,4 +153,28 @@ dataSources:
...
+        - name: SingleRequestProxyFactory
+          file: ./abis/SingleRequestProxyFactory.json
+      eventHandlers:
+        - event: ERC20SingleRequestProxyCreated(address,address,address,indexed bytes,address,uint256,address)
+          handler: handleERC20SingleRequestProxyCreated
+          receipt: true
+        - event: EthereumSingleRequestProxyCreated(address,address,indexed bytes,address,uint256,address)
+          handler: handleEthereumSingleRequestProxyCreated
+          receipt: true
+      file: ./src/singleRequestProxyFactory.ts
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply subgraph.avalanche.yaml:179]: Resolve as No Problem :ok_person:

File: src/singleRequestProxyFactory.ts [hunk 3]
```diff
@@ -24,12 +27,15 @@ export function handleERC20SingleRequestProxyCreated(
   event: ERC20SingleRequestProxyCreated,
 ): void {
   let deployment = new SingleRequestProxyDeployment(
-    event.transaction.hash.toHexString(),
+    event.transaction.hash.toHexString() + "-" + event.logIndex.toString(),
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/singleRequestProxyFactory.ts:30]: Resolve as Won't Fix :no_good:

File: src/singleRequestProxyFactory.ts [hunk 4]
```diff
@@ -8,11 +8,14 @@ export function handleEthereumSingleRequestProxyCreated(
   event: EthereumSingleRequestProxyCreated,
 ): void {
   let deployment = new SingleRequestProxyDeployment(
-    event.transaction.hash.toHexString(),
+    event.transaction.hash.toHexString() + "-" + event.logIndex.toString(),
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/singleRequestProxyFactory.ts:11]: Resolve as Won't Fix :no_good:

File: subgraph.bsc.yaml [hunk 5]
```diff
@@ -153,4 +153,28 @@ dataSources:
...
+        - name: SingleRequestProxyFactory
+          file: ./abis/SingleRequestProxyFactory.json
+      eventHandlers:
+        - event: ERC20SingleRequestProxyCreated(address,address,address,indexed bytes,address,uint256,address)
+          handler: handleERC20SingleRequestProxyCreated
+          receipt: true
+        - event: EthereumSingleRequestProxyCreated(address,address,indexed bytes,address,uint256,address)
+          handler: handleEthereumSingleRequestProxyCreated
+          receipt: true
+      file: ./src/singleRequestProxyFactory.ts
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply subgraph.bsc.yaml:179]: In this case, we don't need replay protection. Duplicates are okay :ok_person:  Resolving as No Probem :ok_person:

File: subgraph.celo.yaml [hunk 6]
```diff
@@ -176,4 +176,28 @@ dataSources:
...
+        - name: SingleRequestProxyFactory
+          file: ./abis/SingleRequestProxyFactory.json
+      eventHandlers:
+        - event: ERC20SingleRequestProxyCreated(address,address,address,indexed bytes,address,uint256,address)
+          handler: handleERC20SingleRequestProxyCreated
+          receipt: true
+        - event: EthereumSingleRequestProxyCreated(address,address,indexed bytes,address,uint256,address)
+          handler: handleEthereumSingleRequestProxyCreated
+          receipt: true
+      file: ./src/singleRequestProxyFactory.ts
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply subgraph.celo.yaml:202]: Resolve as No Problem :ok_person:

File: subgraph.mantle.yaml [hunk 7]
```diff
@@ -86,3 +86,28 @@ dataSources:
           handler: handleTransferWithReferenceAndFee
           receipt: true
       file: ./src/ethFeeProxy.ts
+  - kind: ethereum/contract
+    name: SingleRequestProxyFactory
+    network: mantle
+    source:
+      address: "0xf8cACE7EE4c03Eb4f225434B0709527938D365b4"
+      abi: SingleRequestProxyFactory
+      startBlock: 71485828
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply subgraph.mantle.yaml:95]: No, this start block is correct. Resolving as No Problem :ok_person:

**Review by MantisClone — APPROVED** [review]

**Comment by MantisClone** [comment]
@aimensahnoun deployed to Alchemy. I promoted to Live, manually

Github Action deployed to Subgraph Studio. I published to the decentralized network, manually.

I manually deployed to 0xGraph for Mantle using the following commands:

```
mantisclone@mantisclone-Inspiron-7620:~/projects/payments-subgraph$ graph auth https://subgraph-api.mantle.xyz/deploy
What is the deploy key?: *************************
Deploy key set for https://subgraph-api.mantle.xyz/deploy
mantisclone@mantisclone-Inspiron-7620:~/projects/payments-subgraph$ graph deploy requestnetwork/request-payments-mantle --node https://subgraph-api.mantle.xyz/deploy --ipfs https://subgraph-api.mantle.xyz/ipfs ./subgraph.mantle.yaml
Which version label to use? (e.g. "v0.0.1"): 541f68f
  Skip migration: Bump mapping apiVersion from 0.0.1 to 0.0.2
  Skip migration: Bump mapping apiVersion from 0.0.2 to 0.0.3
  Skip migration: Bump mapping apiVersion from 0.0.3 to 0.0.4
  Skip migration: Bump mapping apiVersion from 0.0.4 to 0.0.5
  Skip migration: Bump mapping apiVersion from 0.0.5 to 0.0.6
  Skip migration: Bump manifest specVersion from 0.0.1 to 0.0.2
  Skip migration: Bump manifest specVersion from 0.0.2 to 0.0.4
✔ Apply migrations
✔ Load subgraph from subgraph.mantle.yaml
  Compile data source: ERC20Proxy => build/ERC20Proxy/ERC20Proxy.wasm
  Compile data source: ERC20FeeProxy_0_2_0 => build/ERC20FeeProxy_0_2_0/ERC20FeeProxy_0_2_0.wasm
  Compile data source: EthProxy_0_3_0 => build/EthProxy_0_3_0/EthProxy_0_3_0.wasm
  Compile data source: EthFeeProxy_0_2_0 => build/EthFeeProxy_0_2_0/EthFeeProxy_0_2_0.wasm
  Compile data source: SingleRequestProxyFactory => build/SingleRequestProxyFactory/SingleRequestProxyFactory.wasm
✔ Compile subgraph
  Copy schema file build/schema.graphql
  Write subgraph file build/ERC20Proxy/abis/ERC20Proxy.json
  Write subgraph file build/ERC20FeeProxy_0_2_0/abis/ERC20FeeProxy-0.2.0.json
  Write subgraph file build/EthProxy_0_3_0/abis/EthProxy-0.3.0.json
  Write subgraph file build/EthFeeProxy_0_2_0/abis/EthFeeProxy-0.2.0.json
  Write subgraph file build/SingleRequestProxyFactory/abis/SingleRequestProxyFactory.json
  Write subgraph manifest build/subgraph.yaml
✔ Write compiled subgraph to build/
  Add file to IPFS build/schema.graphql
                .. QmU78hLFV2N2xRmrWJGmzH8fs9wnCT7CGS23F6rMG6YqA8
  Add file to IPFS build/ERC20Proxy/abis/ERC20Proxy.json
                .. QmUXm6CtZHNtUoNS6hxLefAV1h2ythN4f3N7GUWPgQctV2
  Add file to IPFS build/ERC20FeeProxy_0_2_0/abis/ERC20FeeProxy-0.2.0.json
                .. QmRdM2xXFCi6h9PwRsmBHCPupc2MbsiDfg1wkXjZxvcxTD
  Add file to IPFS build/EthProxy_0_3_0/abis/EthProxy-0.3.0.json
                .. QmWyBM7Yh78mBr1VGCsHkHGfChG5JgeZHtHQzV6ZCRgkg6
  Add file to IPFS build/EthFeeProxy_0_2_0/abis/EthFeeProxy-0.2.0.json
                .. QmSsyL9o9ggK2S7KcPXvNjkogBLsXzik3axgsXAx9SjSbJ
  Add file to IPFS build/SingleRequestProxyFactory/abis/SingleRequestProxyFactory.json
                .. QmWwQrFZzioRXDzSbv7J2xahYYJH6bo5iTdR4CWDQp9oND
  Add file to IPFS build/ERC20Proxy/ERC20Proxy.wasm
                .. QmVpEcwGhh1BTgU2sDJUZAbdM8wmsHwXkDffFShKmLF7bn
  Add file to IPFS build/ERC20FeeProxy_0_2_0/ERC20FeeProxy_0_2_0.wasm
                .. Qmck5KM2S6SFAzuvxj5D1CZ7Yspx8JLNHuwXZksubeqH2P
  Add file to IPFS build/EthProxy_0_3_0/EthProxy_0_3_0.wasm
                .. QmT2wr8Jy6CdKfCbjcDJecG6hys39Kp69NyQ3zVuDdD5yh
  Add file to IPFS build/EthFeeProxy_0_2_0/EthFeeProxy_0_2_0.wasm
                .. QmXu7ie2mXPQmQZkW6wgNtYBPrWkG412Agmufh9AxmtDsy
  Add file to IPFS build/SingleRequestProxyFactory/SingleRequestProxyFactory.wasm
                .. QmWm5ZzvRoqJ6kdDAAaSsXmwkeEM2zXf57ZhoSHrx34Rn6
✔ Upload subgraph to IPFS

Build completed: QmPfW3icbGwAotufrw7ytnjPGHzh3g6ocaASybvYvnz1BJ

Deployed to /api/public/555176e7-c1f4-49f9-9180-f2f03538b039/subgraphs/requestnetwork/request-payments-mantle/541f68f/gn

Subgraph endpoints:
Queries (HTTP):     /api/public/555176e7-c1f4-49f9-9180-f2f03538b039/subgraphs/requestnetwork/request-payments-mantle/541f68f/gn

```

**Pull request merged**